packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,213 @@
+              },
+            },
+          },
+        },
+      }
+
+      return openapiBody
+    }
+  } else if (contentType && knownContentTypes[hoppRequestBody.contentType]) {
+    // treat them as plain text
+    const openapiBody: OpenAPIV3.RequestBodyObject = {
+      content: {
+        ["text/plain"]: {
+          schema: {
+            type: "string",
+            default: hoppRequestBody.body,
+          },
+        },
+      },
+    }
+
+    return openapiBody
+  }
+}
+
+function isValidMethod(method: string): method is OpenAPIV3.HttpMethods {
+  return [
+    "get",
+    "post",
+    "patch",
+    "delete",
+    "options",
+    "head",
+    "patch",
+    "trace",
+  ].includes(method)
+}
+
+const generateOpenApiDocument = (
+  paths: OpenAPIV3.PathsObject
+): OpenAPIV3.Document => ({
+  openapi: "3.0.0",
+  info: {
+    title: "Hoppscotch Openapi Export",
+    version: "1.0.0",
+  },
+  paths,
+})
+
+const generateOpenApiPathFromRequest = (
+  request: HoppRESTRequest
+): E.Either<string, OpenAPIV3.PathItemObject & { pathname: string }> =>
+  pipe(
+    E.Do,
+    E.bind("method", () =>
+      pipe(
+        request.method.toLowerCase(),
+        E.fromPredicate(isValidMethod, () => "INVALID_METHOD")
+      )
+    ),
+    E.bind("origin", () =>
+      pipe(
+        E.tryCatch(
+          () => new URL(request.endpoint),
+          () => "INVALID_URL"
+        )
+      )
+    ),
+    E.bindW(
+      "basePathObject",
+      ({
+        method,
+        origin,
+      }): E.Either<never, OpenAPIV3.PathItemObject & { pathname: string }> =>
+        pipe(
+          {
+            pathname: origin.pathname,
+            [method]: {
+              description: request.name,
+              responses: {
+                200: {
+                  description: "",
+                },
+              },
+            },
+            servers: [
+              {
+                url: origin.host,
+              },
+            ],
+          },
+          E.right
+        )
+    ),
+    E.chainW(({ basePathObject, method }) =>
+      pipe(
+        E.tryCatch(
+          () => pipe(request.body, generateOpenApiRequestBody),
+          () => "INVALID_BODY"
+        ),
+        E.map((openApiBody) => ({
+          ...basePathObject,
+          [method]: {
+            ...basePathObject[method],
+            requestBody: openApiBody,
+          },
+        }))
+      )
+    )
+  )
+
+export const convertHoppToOpenApiCollection = (
+  collections: HoppCollection<HoppRESTRequest>[]
+): E.Either<string, OpenAPIV3.Document> =>
+  pipe(
+    collections,
+    A.map((collection) =>
+      pipe(collection.requests, A.map(generateOpenApiPathFromRequest))
+    ),
+    A.flatten,
+    E.sequenceArray,
+    E.map(RA.toArray),
+    E.chainW((paths) =>
+      pipe(
+        paths,
+        A.reduce({}, (allPaths: OpenAPIV3.PathsObject, path) => ({
+          ...allPaths,
+          [path.pathname]: omit(path, "pathname"),
+        })),
```

Comment:
Can be simplified into a flow

```suggestion
    E.chainW(
      flow( // import { flow } from "fp-ts/function"
        paths,
        A.reduce({}, (allPaths: OpenAPIV3.PathsObject, path) => ({
          ...allPaths,
          [path.pathname]: omit(path, "pathname"),
        })),
```